doc/examples/defining_surfaces.md

```diff
@@ -88,96 +87,94 @@ gs.plot()
 
 ```{code-cell} ipython3
-p0 = polygons.regular_ngon(12,field=AA)
-p1 = polygons.regular_ngon(3,field=AA)
+from flatsurf import polygons
+
+p0 = polygons.regular_ngon(12, field=AA)
+p1 = polygons.regular_ngon(3, field=AA)
 ```
 
 ```{code-cell} ipython3
-p0.plot()+p1.plot()
+p0.plot() + p1.plot()
 ```
 
 The vertices of n-gons are numbered by $\{0,...,n-1\}$, with the $0$-th vertex at the origin. Edge $i$ joins vertex $i$ to vertex $i+1 \pmod{n}$.
 
 We can act on polygon with $2 \times 2$ matrices. We define the rotation by $\frac{\pi}{6}$ below:
 
 ```{code-cell} ipython3
-R = matrix(AA,[[cos(pi/6),-sin(pi/6)],[sin(pi/6),cos(pi/6)]])
+R = matrix(AA, [[cos(pi / 6), -sin(pi / 6)], [sin(pi / 6), cos(pi / 6)]])
 show(R)
 ```
 
 ```{code-cell} ipython3
-R*p1
+R * p1
 ```
 
 Define a surface over the field <code>AA</code> of algebraic reals.
 
 ```{code-cell} ipython3
-surface = Surface_dict(base_ring=AA)
+from flatsurf import MutableOrientedSimilaritySurface
+
+surface = MutableOrientedSimilaritySurface(AA)
 ```
 
 Add two polygons to the surface with labels 0 and 1:
 
 ```{code-cell} ipython3
-surface.add_polygon(p0,label=0)
-```
-
-```{code-cell} ipython3
-surface.add_polygon(p1,label=1)
+surface.add_polygon(p0, label=0)
 ```
 
-Set the "base label" for the surface. This is just a choice of a favorite polygon label.
-
 ```{code-cell} ipython3
-surface.change_base_label(0)
+surface.add_polygon(p1, label=1)
 ```
 
 Glue the edges of polygon 0 to the parallel edges of polygon 1.
 
 ```{code-cell} ipython3
-surface.change_edge_gluing(0,6,1,0)
-surface.change_edge_gluing(0,10,1,1)
-surface.change_edge_gluing(0,2,1,2)
+surface.glue((0, 6), (1, 0))
+surface.glue((0, 10), (1, 1))
+surface.glue((0, 2), (1, 2))
 ```
 
 Add three more rotated triangles and glue them appropriately.
 
 ```{code-cell} ipython3
-for i in range(1,4):
-    surface.add_polygon((R**i)*p1,label=i+1)
-    surface.change_edge_gluing(0,6+i,i+1,0)
-    surface.change_edge_gluing(0,(10+i)%12,i+1,1)
-    surface.change_edge_gluing(0,2+i,i+1,2)
+for i in range(1, 4):
+    surface.add_polygon((R**i) * p1, label=i + 1)
+    surface.glue((0, 6 + i), (i + 1, 0))
+    surface.glue((0, (10 + i) % 12), (i + 1, 1))
+    surface.glue((0, 2 + i), (i + 1, 2))
 ```
 
-Now we have a closed surface. In fact we have defined a Translation Surface. The package also supports
-SimilaritySurface, ConeSurface, HalfDilationSurface, DilationSurface, and HalfTranslationSurface.
+Now we have a closed surface. In fact this is a translation surface:
 
 ```{code-cell} ipython3
-s=TranslationSurface(surface)
+surface
 ```
 
-Test to insure that we created the translation surface correctly. (Errors would be printed if you did not glue parallel edges, or have some unglued edges, etc.)
+Once we are done building the surface, it is recommended to make the surface immutable. This lets sage-flatsurf speed up many operations on the surface and makes it possible to compute invariants such as the stratum:
```

Comment:
Because the `stratum()` function is implemented in the category.

We cannot refine the category of a mutable surface. Once we make it a translation surface, it will be at least a translation surface forever.

I could explain how to call it anyway when needed without making the surface immutable.

Reply:
I added more documentation in https://github.com/flatsurf/sage-flatsurf/pull/220/commits/d162bfe369db056ea5713a24b64dbfa7a5ddf9fb. Feel free to have a look.